controllers/metric_controller.go

```diff
@@ -36,11 +37,14 @@ import (
 const (
 	// MetricController is the name of this controller in logs and other outputs.
 	MetricController = "metrics"
+	// monitoringLabel is the label added to the watch namespace to indicate cluster monitoring is enabled.
+	monitoringLabel = "openshift.io/cluster-monitoring"
 )
 
 type metricReconciler struct {
 	*monclient.MonitoringV1Client
 	instanceReconciler
+	prevState map[string]string
```

Comment:
Can't this just be `monitoringEnabled bool`?
The namespace really shouldn't be changing, that requires removing and reinstalling the operator into a different namespace.
Having this be a map is adding a bit of unneeded complexity.